src/components/searchBar/index.tsx

```diff
@@ -1,23 +1,86 @@
 import React, { useCallback, useState } from "react";
 
+interface SearchResultListProps {
+  id: number;
+  name: string;
+}
+
 export default function SearchBar() {
   const [value, setValue] = useState("");
+  const [searchResultList, setSearchResultList] = useState<SearchResultListProps[]>([]);
+  const [selectedIngredientsList, setSelectedIngredientsList] = useState<SearchResultListProps[]>([]);
 
   const handleChangeInput = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       event.preventDefault();
       setValue(event.target.value);
+      fetch(
+        `https://api.spoonacular.com/food/ingredients/autocomplete?apiKey=${process.env.NEXT_PUBLIC_SPOONACULAR_API_KEY}&query=${event.target.value}&number=10`
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          setSearchResultList(result);
+        });
     },
     []
   );
 
+  const handleSelectIngredient = useCallback(
+    (ingredient: SearchResultListProps) => {
+      setSelectedIngredientsList((current) => [
+        ...current,
+        {
+          ...ingredient,
```

Comment:
This is totally subjective
```suggestion
      setSelectedIngredientsList((prev) => [
        ...prev,
        {
          ...ingredient,
```